packages/flutter/lib/src/material/page.dart

```diff
@@ -88,12 +89,25 @@ class AnimationMean extends CompoundAnimation<double> {
 // is fully on-screen. 0.0 is fully off-screen to the right. 1.0 is off-screen
 // to the left.
 class _CupertinoTransitionCurve extends Curve {
-  _CupertinoTransitionCurve();
+  _CupertinoTransitionCurve(this.curve);
+
+  Curve curve;
 
   @override
   double transform(double t) {
-    if (t > 0.5)
-      return (t - 0.5) / 3.0 + 0.5;
+    // The range is divided in 2 halves: [0.0, 0.5] applies to the top-most
+    // route, while [0.5, 1.0] applies to the one underneath it. We scale both
+    // ranges to [0.0, 1.0] when applying transformations.
```

Comment:
Is this any clearer:

    // The input [t] is the average of the previous and current route's
    // animations. So, when [t] is in the range [0.0, 0.5], we're dealing with
    // the top-most route. Likewise, [0.5, 1.0] applies to the route underneath
    // it (to the left of the screen).
    // To make the curves apply properly, we scale [t] to the range [0.0, 1.0]
    // before applying transformations.